drivers/serial/uart_mcux_flexcomm.c

```diff
@@ -342,6 +379,636 @@ static int mcux_flexcomm_uart_config_get(const struct device *dev,
+	data->tx_data.active_block.dest_address = (uint32_t) &config->base->FIFOWR;
+	data->tx_data.active_block.block_size = len;
+	data->tx_data.active_block.next_block = NULL;
+
+	ret = dma_config(config->tx_dma.dev, config->tx_dma.channel,
+				(struct dma_config *) &config->tx_dma.cfg);
+	if (ret) {
+		return ret;
+	}
+
+	/* Enable interrupt for when TX fifo is empty (all data transmitted) */
+	config->base->FIFOINTENSET |= USART_FIFOINTENSET_TXLVL_MASK;
+
+	/* Enable TX DMA requests */
+	USART_EnableTxDMA(config->base, true);
+
+	/* Trigger the DMA to start transfer */
+	ret = dma_start(config->tx_dma.dev, config->tx_dma.channel);
+	if (ret) {
+		return ret;
+	}
+
+	/* Schedule a TX abort for @param timeout */
+	if (timeout != SYS_FOREVER_US) {
+		k_work_schedule(&data->tx_data.timeout_work, K_USEC(timeout));
+	}
+
+	irq_unlock(key);
+
+	return ret;
+}
+
+static int mcux_flexcomm_uart_tx_abort(const struct device *dev)
+{
+	const struct mcux_flexcomm_config *config = dev->config;
+	struct mcux_flexcomm_data *data = dev->data;
+	int ret = 0;
+
+	/* First disable DMA requests from UART to prevent transfer
+	 * status change during the abort routine
+	 */
+	USART_EnableTxDMA(config->base, false);
+
+	/* In case there is no transfer to abort */
+	if (data->tx_data.xfer_len == 0) {
+		return -EFAULT;
+	}
+
+	/* In case a user called this function, do not abort twice */
+	(void)k_work_cancel_delayable(&data->tx_data.timeout_work);
+
+	/* Getting dma status to use to calculate bytes sent */
+	struct dma_status status = {0};
+
+	ret = dma_get_status(config->tx_dma.dev, config->tx_dma.channel, &status);
+	if (ret < 0) {
+		return ret;
+	}
+
+	/* Done with the DMA transfer, can stop it now */
+	ret = dma_stop(config->tx_dma.dev, config->tx_dma.channel);
+	if (ret) {
+		return ret;
+	}
+
+	/* Define TX abort event before resetting driver variables */
+	size_t sent_len = data->tx_data.xfer_len - status.pending_length;
+	const uint8_t *aborted_buf = data->tx_data.xfer_buf;
+	struct uart_event tx_abort_event = {
+		.type = UART_TX_ABORTED,
+		.data.tx.buf = aborted_buf,
+		.data.tx.len = sent_len
+	};
+
+	/* Driver data needs reset since there is no longer an ongoing
+	 * transfer, this should before the user callback, not after,
+	 * just in case the user callback calls tx again
+	 */
+	data->tx_data.xfer_len = 0;
+	data->tx_data.xfer_buf = NULL;
+
+	async_user_callback(dev, &tx_abort_event);
+
+	return ret;
+}
+
+static int mcux_flexcomm_uart_rx_enable(const struct device *dev, uint8_t *buf,
+					const size_t len, const int32_t timeout)
+{
+	const struct mcux_flexcomm_config *config = dev->config;
+	struct mcux_flexcomm_data *data = dev->data;
+	int ret = 0;
+
+	if (config->rx_dma.dev == NULL) {
+		return -ENODEV;
+	}
+
+	/* Getting DMA status to tell if channel is busy or not set up */
+	struct dma_status status;
+
+	ret = dma_get_status(config->rx_dma.dev, config->rx_dma.channel, &status);
+
+	if (ret < 0) {
+#ifdef CONFIG_DMA_MCUX_LPC
```

Comment:
Same comment as above about DMA driver specific behavior, but in this case using `data->rx_data.xfer_buf`